agent/bench-scripts/pbench-user-benchmark

```diff
@@ -105,6 +116,9 @@ echo $iteration >> $benchmark_iterations
 $benchmark_bin 2>&1 | tee $benchmark_results_dir/result.txt
 pbench-stop-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
 pbench-postprocess-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
-pbench-collect-sysinfo --group=$tool_group --dir=$benchmark_run_dir end
-
+if [[ -n "$collect_sysinfo" ]]; then
```

Comment:
I don't think we need to add an `if` statement here.  No matter what the requested behavior is, we can still invoke `pbench-collect-sysinfo` and let it determine what its behavior should be.